modeshape-jcr/src/main/java/org/modeshape/jcr/shell/cmd/Command.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.jcr.shell.cmd;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.modeshape.jcr.shell.Interpreter;
+
+/**
+ *
+ * @author kulikov
+ */
+public abstract class Command {
```

Comment:
can this class (or any of its subclasses) ever be used from multiple threads ? i.e. is thread safety an issue to consider or not (I'm asking because there seems to be a lot of state held)